ambry-quota/src/main/java/com/github/ambry/quota/capacityunit/AmbryCUQuotaEnforcer.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2022 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.quota.capacityunit;
+
+import com.github.ambry.config.QuotaConfig;
+import com.github.ambry.messageformat.BlobInfo;
+import com.github.ambry.quota.QuotaEnforcer;
+import com.github.ambry.quota.QuotaException;
+import com.github.ambry.quota.QuotaName;
+import com.github.ambry.quota.QuotaRecommendation;
+import com.github.ambry.quota.QuotaResource;
+import com.github.ambry.quota.QuotaSource;
+import com.github.ambry.quota.QuotaUtils;
+import com.github.ambry.rest.RestRequest;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Callable;
+
+
+/**
+ * Implementation of {@link QuotaEnforcer} for CU quota of Ambry's {@link QuotaResource}s.
+ * Note that this implementation is not thread safe or atomic. The caller should take care of thread safety guarantees
+ * where needed.
+ */
+public class AmbryCUQuotaEnforcer implements QuotaEnforcer {
+  private final static List<QuotaName> SUPPORTED_QUOTA_NAMES =
+      Collections.unmodifiableList(Arrays.asList(QuotaName.READ_CAPACITY_UNIT, QuotaName.WRITE_CAPACITY_UNIT));
+  private final static long NO_THROTTLE_RETRY_AFTER_MS = -1;
+  private final QuotaSource quotaSource;
+  private final float maxFrontendCuUsageToAllowExceed;
+  private final long throttleRetryAfterMs;
+
+  /**
+   * Constructor for {@link AmbryCapacityUnitQuotaEnforcer}.
+   * @param quotaSource {@link QuotaSource} where the quota limit and usage will be saved and retrieved from.
+   */
+  public AmbryCUQuotaEnforcer(QuotaSource quotaSource, QuotaConfig quotaConfig) {
+    this.quotaSource = quotaSource;
+    this.throttleRetryAfterMs = quotaConfig.cuQuotaAggregationWindowInSecs;
+    this.maxFrontendCuUsageToAllowExceed = quotaConfig.maxFrontendCuUsageToAllowExceed;
+  }
+
+  @Override
+  public void init() {
+  }
+
+  @Override
+  public void charge(RestRequest restRequest, BlobInfo blobInfo, Map<QuotaName, Double> requestCostMap)
```

Comment:
nit: i think we might be able to get rid of BlobInfo in the function signature, since it's not really used anywhere.